java/org/contikios/cooja/Simulation.java

```diff
@@ -696,55 +748,39 @@ public boolean setConfigXML(Element root, boolean quick) throws Exception {
    *          Mote to remove
    */
   public void removeMote(final Mote mote) {
+    invokeSimulationThread(() -> doRemoveMote(mote));
+  }
 
-    /* Simulation is running, remove mote in simulation loop */
-    Runnable removeMote = new Runnable() {
-      @Override
-      public void run() {
-        motes.remove(mote);
-        currentRadioMedium.unregisterMote(mote, Simulation.this);
+  private void doRemoveMote(Mote mote) {
+    motes.remove(mote);
 
-        /* Dispose mote interface resources */
-        mote.removed();
-        for (MoteInterface i: mote.getInterfaces().getInterfaces()) {
-          i.removed();
-        }
-
-        setChanged();
-        notifyObservers(mote);
+    if (currentRadioMedium != null) {
+      currentRadioMedium.unregisterMote(mote, Simulation.this);
+    }
 
-        // Delete all events associated with deleted mote.
-        eventQueue.removeIf(
-          (TimeEvent ev) ->
-            ev instanceof MoteTimeEvent && ((MoteTimeEvent)ev).getMote() == mote);
-      }
-    };
+    /* Dispose mote interface resources */
+    mote.removed();
 
-    if (!isRunning()) {
-      /* Simulation is stopped, remove mote immediately */
-      removeMote.run();
-    } else {
-      /* Remove mote from simulation thread */
-      invokeSimulationThread(removeMote);
+    for (MoteInterface i: mote.getInterfaces().getInterfaces()) {
+      i.removed();
     }
 
+    // Delete all events associated with deleted mote.
+    eventQueue.removeIf(ev -> ev instanceof MoteTimeEvent moteTimeEvent && moteTimeEvent.getMote() == mote);
+
     getCooja().closeMotePlugins(mote);
+
+    setChanged();
+    notifyObservers(mote);
```

Comment:
This changes the order of events and notifications. What is the protocol inside Cooja, should the notification come after the thing has happened?

Reply:
I missed this and will restore the order. Thanks!